WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShipments.kt

```diff
@@ -80,4 +90,20 @@ class GetShipments @Inject constructor(
             shipmentUIModel.copy(purchased = true, label = mapper.invoke(noRefundedLabelForShipment))
         }
     }
+
+    private fun mergeDestinationAddresses(
+        shipmentUIModelList: List<ShipmentUIModel>,
+        destinationAddresses: Map<String, DestinationAddressDTO>
+    ) = shipmentUIModelList.map { shipmentUIModel ->
+        val id = shipmentUIModel.remoteId
+        if (id != null && destinationAddresses.contains(getStoredDataKey(id))) {
+            destinationAddresses[getStoredDataKey(id)]?.let {
+                shipmentUIModel.copy(label = shipmentUIModel.label?.copy(destinationAddress = mapper.invoke(it)))
+            } ?: shipmentUIModel
+        } else {
+            shipmentUIModel
+        }
```

Comment:
Np, you can probably remove redundant checks here. Either: 

```suggestion
        if (id != null && destinationAddresses.contains(getStoredDataKey(id))) {
            destinationAddresses[getStoredDataKey(id)].let {
                shipmentUIModel.copy(label = shipmentUIModel.label?.copy(destinationAddress = mapper.invoke(it)))
            }
        } else {
            shipmentUIModel
        }
```

Or 

```suggestion
        if (id != null) {
            destinationAddresses[getStoredDataKey(id)]?.let {
                shipmentUIModel.copy(label = shipmentUIModel.label?.copy(destinationAddress = mapper.invoke(it)))
            } ?: shipmentUIModel
        } else {
            shipmentUIModel
        }
```

I kind of lean more towards the second option to avoid having to traverse the map twice. Wdyt?

Reply:
Good poin! I committed your second suggestion.